### ISSUE TITLE
Can't summarize min/max on a text field in the query builder

### ISSUE THREAD

**Issue by ariya** [metabase/metabase#18207]
**Describe the bug**

Using `max` on a text field doesn't work in the query builder.
Yet, running the SQL version in the native query works as expected.

**To Reproduce**

1. Ask a question, Custom question.
2. Sample Dataset, Products.
3. Summarize, Custom Expression, and type `Max([Vendor])` and name it as `LastVendor`.
4. Group by Category.
5. Visualize.

**Expected**

A table like this:

![image](https://user-images.githubusercontent.com/7288/135698812-5ea2047e-891b-49ed-9fc1-273042a4824a.png)

**Actual**

A table displaying the Products with no grouping/summary at all (just a regular table).

**However**, if after Step 4 we convert the question to SQL:

```sql
SELECT "PUBLIC"."PRODUCTS"."CATEGORY" AS "CATEGORY", max("PUBLIC"."PRODUCTS"."VENDOR") AS "LastVendor"
FROM "PUBLIC"."PRODUCTS"
GROUP BY "PUBLIC"."PRODUCTS"."CATEGORY"
ORDER BY "PUBLIC"."PRODUCTS"."CATEGORY" ASC
```

then the query gives the expected result.

**Comment by flamber** [comment]
Historically related to https://github.com/metabase/metabase/issues/4482#issuecomment-927821792 - this might get a upped to P2 at some point.

**Comment by flamber** [comment]
Fixed by #19416 in 0.42
